### PR TITLE
Add several missing inspectRuntimeShape() calls

### PR DESCRIPTION
**Pull request by leventov**

### CODE REVIEW

File: extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java [hunk 1]
```diff
@@ -181,4 +182,9 @@ private static int smear(int hashCode)
     return hashCode ^ (hashCode >>> 7) ^ (hashCode >>> 4);
   }
 
+  @Override
+  public void inspectRuntimeShape(RuntimeShapeInspector inspector)
+  {
+    inspector.visit("selector", selector);
```
Comment (leventov) [inline extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java]: @AlexanderSaydakov could you review this? Do parameters of `HllSketch`, `lgK` and `tgtHllType` affect the runtime shape of `aggregate()`?

Comment (drcrallen) [reply extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java]: If I recall the original design spec for monomorphic processing, missing some fields results in sub-optimal performance, but no worse than specifying no runtime shape. Is that still true, as in, could it be merged as is and confirmed later with no bad side-effect?

Comment (leventov) [reply extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java]: @AlexanderSaydakov runtime shape could be intuited as the actual hot machine code path when executing the `@CalledFromHotLoop` method, `aggregate()` in this example. E. g. if `lgK` just participates some mathematical expressions, the hot machine code will be something like `mul %ax, %bx` no matter what value is in `lgK`. This means that `lgK` does *not* affect the runtime shape. On the contrary, if the internal logic is something like:
```java
if (lgK < 5) {
  execute some logic
} else {
  execute some entirely different logic
}
```
Then `lgK` does affect the hot code path, and therefore the runtime shape.

Comment (leventov) [reply extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/HllSketchBuildBufferAggregator.java]: @AlexanderSaydakov thank you. I included `lgK` in runtime shaped of both `HllSketchBuildBufferAggregator` and `HllSketchMergeBufferAggregator`, assuming that different paths exist in both `HllSketch.update()` and `Union.update()`. See https://github.com/apache/incubator-druid/pull/6893/commits/fb7ee63d9c0186c365a0b7b3d1c8cadc5e025168.

**Comment by AlexanderSaydakov** [comment]
What is the cost of making a mistake both ways?
Do I understand correctly the above comment by drcrallen that missing parameters that do affect the runtime shape would miss a potential performance improvement? What about adding unnecessary ones? Any possibility to affect correctness of the results?

**Comment by leventov** [comment]
@AlexanderSaydakov the cost of not inspecting something that influences the runtime shape could be significant or not very significant, depending on how big and where in the call stack is that influence. In the worst case, it cancels the optimization framework for the given type of aggregator.

There might also be some performance cost of inspecting something extra, but it is generally lower than the cost of not inspecting something important:
 - Creation of extra unneeded classes, storing their bytecode and native code in memory (hence extra icache pollution)
 - Separate runtime shapes (those that in fact shouldn't be separate) don't pass the optimization threshold individually, but if there was just one, it would pass the threshold. Bad, but if the runtime shape barely exceeds the optimization threshold not optimizing it is not a super big deal.

None of the above affects the correctness of results whatsoever.

**Pull request merged**